src/directives/jf_codemirror/jf_codemirror.js

```diff
@@ -39,10 +40,15 @@ class jfCodeController {
     $onInit() {
 	    this._formatModel();
 	    this.autofocus = this.autofocus === 'true';
+        let readOnlyParam = !this.allowEdit;
```

Comment:
I would rename `readOnlyParam` to something a but more self explanatory as `isReadOnlyMode` .